sdks/js/DittoChatCore/src/slices/useMessages.ts

```diff
@@ -180,290 +407,80 @@ export const createMessageSlice: CreateSlice<MessageSlice> = (
       }
     },
 
-    async createMessage(room: Room, text: string) {
-      if (!ditto) return;
-      if (!userId) return;
-
-      try {
-        // Fetch current user to get name
-        const currentUserResult = await ditto.store.execute<Room>(
-          `SELECT * FROM ${userCollectionKey} WHERE _id = :id`,
-          { id: userId },
-        );
-        const userValue = currentUserResult.items?.[0]?.value;
-        const fullName = userValue?.name ?? userId;
-
-        const roomResult = await ditto.store.execute(
-          `SELECT * FROM ${room.collectionId || "rooms"} WHERE _id = :id`,
-          { id: room._id },
-        );
-
-        if (roomResult.items.length === 0) {
-          console.error("Room not found:", room._id);
-          return;
-        }
+    async createMessage(room: Room, text: string, mentions: Mention[] = []) {
+      if (!ditto || !userId) return;
 
-        const actualRoom = roomResult.items[0].value;
-        const messagesId = actualRoom.messagesId;
+      // Check mention permission if mentions are provided
+      if (mentions.length > 0 && !_get().canPerformAction("canMentionUsers")) {
+        console.warn("Permission denied: canMentionUsers is false");
+        // Create message without mentions
+        mentions = [];
```

Comment:
Can we refactor to avoid mutating function arguments? E.g.

```js
const filteredMentions = mentions.length > 0 && _get().canPerformAction("canMentionUsers") 
  ? mentions 
  : [];
```